things fixed:
 - respect env $(CC)
 - do not override -O settings in env CFLAGS with -O2
 - drop useless (to us) -I/-L paths -- let toolchain find upnp
 - use standard LDLIBS rather than LIBS variable
 - respect env LDFLAGS
 - respect env CPPFLAGS
 - drop pointless implicit %.o rule
 - fix install target
   - respect DESTDIR
   - create paths
   - use proper permissions
   - install used gif files

we add -D_GNU_SOURCE to get strnlen() prototype out of glibc

--- linuxigd-1.0/Makefile
+++ linuxigd-1.0/Makefile
@@ -3,11 +3,10 @@
 #LIBIPTC_PREFIX=/usr
 
-CC=gcc
-INCLUDES= -I$(LIBUPNP_PREFIX)/include -I../include
-LIBS= -lpthread -lupnp -lixml -lthreadutil -L$(LIBUPNP_PREFIX)/lib -L../libs
+LDLIBS += -lpthread -lupnp -lixml -lthreadutil
 FILES= main.o gatedevice.o pmlist.o util.o config.o
 
-CFLAGS += -Wall -g -O2
+CFLAGS += -Wall
+CPPFLAGS += -D_GNU_SOURCE
 
 ifdef HAVE_LIBIPTC
 ifdef LIBIPTC_PREFIX
@@ -23,21 +22,18 @@
 all: upnpd
 
 upnpd: $(FILES)
-	$(CC) $(CFLAGS) $(FILES) $(LIBS) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(LDLIBS)
-	@echo "make $@ finished on `date`"
-
-%.o:	%.c
-	$(CC) $(CFLAGS) $(INCLUDES) -c $<
 
 clean:
 	rm -f *.o upnpd
 
 install: upnpd
-	install -d /etc/linuxigd
-	install etc/gatedesc.xml /etc/linuxigd
-	install etc/gateconnSCPD.xml  /etc/linuxigd
-	install etc/gateicfgSCPD.xml /etc/linuxigd
-	install etc/dummy.xml /etc/linuxigd
-	install upnpd $(PREFIX)/sbin
-	install upnpd.8 $(PREFIX)/share/man/man8
-	if [ ! -f /etc/upnpd.conf ]; then install etc/upnpd.conf /etc; fi
+	install -d -m755 $(DESTDIR)/etc/linuxigd $(DESTDIR)$(PREFIX)/sbin $(DESTDIR)$(PREFIX)/share/man/man8
+	install -m644 etc/gatedesc.xml $(DESTDIR)/etc/linuxigd
+	install -m644 etc/ligd.gif $(DESTDIR)/etc/linuxigd
+	install -m644 etc/gateconnSCPD.xml  $(DESTDIR)/etc/linuxigd
+	install -m644 etc/gateicfgSCPD.xml $(DESTDIR)/etc/linuxigd
+	install -m644 etc/dummy.xml $(DESTDIR)/etc/linuxigd
+	install -m755 upnpd $(DESTDIR)$(PREFIX)/sbin
+	install -m644 upnpd.8 $(DESTDIR)$(PREFIX)/share/man/man8
+	install -m644 etc/upnpd.conf $(DESTDIR)/etc
